[fix](hive) sync DDL command to other FE

morningman opened the pull request:
### What problem does this PR solve?

When executing DDL on a hive table, the DDL command should be synced to other FE,
so that other FE can invalidate the cache of that table or db.
Otherwise, after executing the DDL on a hive table on non-master FE and then query that table,
we can still access to the cached files.

Modify the following ddl:
1. create db/table
2. drop db/table
3. truncate table

For these operations, I saved the info like dbname, tablename, and sync them to other FE,
so that other FE can refresh these objects to get latest metadata info

### Release Note

This PR add a new edit log type: `OperationType.OP_NEW_CREATE_DB`.
So that it can't downgrade

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [x] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [x] No.
    - [ ] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

morningman committed:
[fix](hive) sync trucate table command to other FE

Thearas commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

morningman committed:
1

morningman committed:
2

morningman committed:
1

morningman commented:
This pull request includes significant changes to enhance the handling of database and table operations in the `Env` and `ExternalCatalog` classes. The most important changes include adding new methods to replay database and table operations, updating existing methods to use these new methods, and improving logging for these operations.

Enhancements to database and table operations:

* [`fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java`](diffhunk://#diff-bbd4eef320c67a1b2702d6cabf12b794dd8efb6b1bc61e9b70ebc639f4a0ece0R3262-R3272): Added new methods `replayNewCreateDb`, `replayDropDb`, `replayCreateTable`, and `replayTruncateTable` to handle replaying of database and table operations. Updated existing methods to use these new methods for better modularity and clarity. [[1]](diffhunk://#diff-bbd4eef320c67a1b2702d6cabf12b794dd8efb6b1bc61e9b70ebc639f4a0ece0R3262-R3272) [[2]](diffhunk://#diff-bbd4eef320c67a1b2702d6cabf12b794dd8efb6b1bc61e9b70ebc639f4a0ece0L3269-R3291) [[3]](diffhunk://#diff-bbd4eef320c67a1b2702d6cabf12b794dd8efb6b1bc61e9b70ebc639f4a0ece0L4149-R4178) [[4]](diffhunk://#diff-bbd4eef320c67a1b2702d6cabf12b794dd8efb6b1bc61e9b70ebc639f4a0ece0R5882-R5891)

* [`fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalCatalog.java`](diffhunk://#diff-50477c5ff1b42452704e942fe5fc738563dc64182690f378925fbc2401be5d48R934-R947): Added methods `replayCreateDb`, `replayDropDb`, `replayCreateTable`, `replayDropTable`, and `replayTruncateTable` to support replaying operations. Enhanced logging in methods `createDb`, `dropDb`, `createTable`, and `truncateTable` to provide more detailed information. [[1]](diffhunk://#diff-50477c5ff1b42452704e942fe5fc738563dc64182690f378925fbc2401be5d48R934-R947) [[2]](diffhunk://#diff-50477c5ff1b42452704e942fe5fc738563dc64182690f378925fbc2401be5d48R957-R970) [[3]](diffhunk://#diff-50477c5ff1b42452704e942fe5fc738563dc64182690f378925fbc2401be5d48L958-R997) [[4]](diffhunk://#diff-50477c5ff1b42452704e942fe5fc738563dc64182690f378925fbc2401be5d48R1007-R1020) [[5]](diffhunk://#diff-50477c5ff1b42452704e942fe5fc738563dc64182690f378925fbc2401be5d48L1079-R1136)

* [`fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetadataOps.java`](diffhunk://#diff-c75ea6d8265f80493833782fd0736d8189d5bec2a91773ce9a7164dc97e88ca1L109-R110): Introduced methods `createDbImpl`, `dropDbImpl`, `createTableImpl`, `dropTableImpl`, `afterCreateDb`, `afterDropDb`, `afterCreateTable`, `afterDropTable`, and `afterTruncateTable` to separate the implementation and post-processing of operations. Updated existing methods to call these new methods. [[1]](diffhunk://#diff-c75ea6d8265f80493833782fd0736d8189d5bec2a91773ce9a7164dc97e88ca1L109-R110) [[2]](diffhunk://#diff-c75ea6d8265f80493833782fd0736d8189d5bec2a91773ce9a7164dc97e88ca1L132-R145) [[3]](diffhunk://#diff-c75ea6d8265f80493833782fd0736d8189d5bec2a91773ce9a7164dc97e88ca1L152-R168) [[4]](diffhunk://#diff-c75ea6d8265f80493833782fd0736d8189d5bec2a91773ce9a7164dc97e88ca1L278-R309) [[5]](diffhunk://#diff-c75ea6d8265f80493833782fd0736d8189d5bec2a91773ce9a7164dc97e88ca1R319-R332)

* [`fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergMetadataOps.java`](diffhunk://#diff-7cf5a22dfe9d3e82d07a651021cae7a8764b47b2fbfcbbf21737888adaf0c84eL119-R135): Similar to `HiveMetadataOps`, added methods `createDbImpl`, `dropDbImpl`, `createTableImpl`, `dropTableImpl`, `afterCreateDb`, `afterDropDb`, and `afterCreateTable` to handle the implementation and post-processing of database and table operations. [[1]](diffhunk://#diff-7cf5a22dfe9d3e82d07a651021cae7a8764b47b2fbfcbbf21737888adaf0c84eL119-R135) [[2]](diffhunk://#diff-7cf5a22dfe9d3e82d07a651021cae7a8764b47b2fbfcbbf21737888adaf0c84eL152-R159) [[3]](diffhunk://#diff-7cf5a22dfe9d3e82d07a651021cae7a8764b47b2fbfcbbf21737888adaf0c84eR182-R190) [[4]](diffhunk://#diff-7cf5a22dfe9d3e82d07a651021cae7a8764b47b2fbfcbbf21737888adaf0c84eL220-R239)

These changes improve the modularity and maintainability of the code, making it easier to manage and extend the functionality for database and table operations.

morningman commented:
run buildall

morningman committed:
4

morningman commented:
run buildall

morningman committed:
5

morningman committed:
6

morningman committed:
7

morningman commented:
run buildall

morningman committed:
9

morningman commented:
run buildall

morningman committed:
fix

morningman commented:
run buildall
